src/sinks/clickhouse.rs

```diff
@@ -174,18 +175,22 @@ impl RetryLogic for ClickhouseRetryLogic {
         self.inner.is_retriable_error(error)
     }
 
-    fn should_retry_response(&self, response: &Self::Response) -> bool {
-        if response.status() == StatusCode::INTERNAL_SERVER_ERROR {
-            let body = response.body();
-
-            // Currently, clickhouse returns 500's incorrect data and type mismatch errors.
-            // This attempts to check if the body starts with `Code: {code_num}` and to not
-            // retry those errors.
-            //
-            // Reference: https://github.com/timberio/vector/pull/693#issuecomment-517332654
-            !(body.starts_with(b"Code: 117") || body.starts_with(b"Code: 53"))
-        } else {
-            self.inner.should_retry_response(response)
+    fn should_retry_response(&self, response: &Self::Response) -> Option<Cow<str>> {
+        match response.status() {
+            StatusCode::INTERNAL_SERVER_ERROR => {
+                let body = response.body();
+
+                // Currently, clickhouse returns 500's incorrect data and type mismatch errors.
+                // This attempts to check if the body starts with `Code: {code_num}` and to not
+                // retry those errors.
+                //
+                // Reference: https://github.com/timberio/vector/pull/693#issuecomment-517332654
+                match body.starts_with(b"Code: 117") || body.starts_with(b"Code: 53") {
+                    false => Some(format!("{:?}", body).into()),
```

Comment:
What will this look like? If body is bytes, will it be something like `"[87, 62, 93, ...]"`?

Reply:
Good point. I was looking at Elasticsearch, which gave this:
```Sep 05 18:15:31.765  WARN sink{name="elasticsearch"}: vector::sinks::util::retries: retrying after response. reason=500 Internal Server Error: b"{\"error\":{\"root_cause\":[{\"type\":\"json_parse_exception\",\"reason\":\"Illegal character ((CTRL-CHAR, code 31)): only regular white space (\\\\r, \\\\n, \\\\t) is allowed between tokens\\n at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper@2856859; line: 1, column: 2]\"}],\"type\":\"json_parse_exception\",\"reason\":\"Illegal character ((CTRL-CHAR, code 31)): only regular white space (\\\\r, \\\\n, \\\\t) is allowed between tokens\\n at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper@2856859; line: 1, column: 2]\"},\"status\":500}"```

Reply:
Not the prettiest, but that seems reasonable to me.

Reply:
I think if we get some bytes like, in this case, it should be parsed via utf8 lossy then printed.

Reply:
Done